File: backend/src/main/java/com/tutict/finalassignmentbackend/entity/SysDict.java
package com.tutict.finalassignmentbackend.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 数据字典表实体类，对应数据库表 "sys_dict"
 * 用于系统数据字典的配置和管理
 */
@Data
@TableName("sys_dict")
public class SysDict implements Serializable {
    /**
     * 序列化版本 UID
     */
    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 字典 ID，主键，自动增长
     */
    @TableId(value = "dict_id", type = IdType.AUTO)
    private Integer dictId;

    /**
     * 父字典 ID (0表示顶级)
     */
    @TableField("parent_id")
    private Integer parentId;

    /**
     * 字典类型
     */
    @TableField("dict_type")
    private String dictType;

    /**
     * 字典编码
     */
    @TableField("dict_code")
    private String dictCode;

    /**
     * 字典标签
     */
    @TableField("dict_label")
    private String dictLabel;

    /**
     * 字典值
     */
    @TableField("dict_value")
    private String dictValue;

    /**
     * 字典描述
     */
    @TableField("dict_description")
    private String dictDescription;

    /**
     * CSS类名
     */
    @TableField("css_class")
    private String cssClass;

    /**
     * 列表样式
     */
    @TableField("list_class")
    private String listClass;

    /**
     * 是否默认 (1=是, 0=否)
     */
    @TableField("is_default")
    private Boolean isDefault;

    /**
     * 是否固定 (1=是, 0=否)
     */
    @TableField("is_fixed")
    private Boolean isFixed;

    /**
     * 状态 (Active, Inactive)
     */
    @TableField("status")
    private String status;

    /**
     * 排序
     */
    @TableField("sort_order")
    private Integer sortOrder;

    /**
     * 创建时间
     */
    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /**
     * 创建人
     */
    @TableField("created_by")
    private String createdBy;

    /**
     * 更新人
     */
    @TableField("updated_by")
    private String updatedBy;

    /**
     * 软删除时间戳
     */
    @TableField("deleted_at")
    @TableLogic(value = "null", delval = "now()")
    private LocalDateTime deletedAt;

    /**
     * 备注
     */
    @TableField("remarks")
    private String remarks;
}

File: flutter_app/lib/features/model/sys_dict.dart
class SysDictModel {
  final int? dictId;
  final int? parentId;
  final String? dictType;
  final String? dictCode;
  final String? dictLabel;
  final String? dictValue;
  final String? dictDescription;
  final String? cssClass;
  final String? listClass;
  final bool? isDefault;
  final bool? isFixed;
  final String? status;
  final int? sortOrder;
  final DateTime? createdTime;
  final DateTime? modifiedTime;
  final String? createdBy;
  final String? updatedBy;
  final DateTime? deletedAt;
  final String? remarks;

  const SysDictModel({
    this.dictId,
    this.parentId,
    this.dictType,
    this.dictCode,
    this.dictLabel,
    this.dictValue,
    this.dictDescription,
    this.cssClass,
    this.listClass,
    this.isDefault,
    this.isFixed,
    this.status,
    this.sortOrder,
    this.createdTime,
    this.modifiedTime,
    this.createdBy,
    this.updatedBy,
    this.deletedAt,
    this.remarks,
  });

  SysDictModel copyWith({
    int? dictId,
    int? parentId,
    String? dictType,
    String? dictCode,
    String? dictLabel,
    String? dictValue,
    String? dictDescription,
    String? cssClass,
    String? listClass,
    bool? isDefault,
    bool? isFixed,
    String? status,
    int? sortOrder,
    DateTime? createdTime,
    DateTime? modifiedTime,
    String? createdBy,
    String? updatedBy,
    DateTime? deletedAt,
    String? remarks,
  }) {
    return SysDictModel(
      dictId: dictId ?? this.dictId,
      parentId: parentId ?? this.parentId,
      dictType: dictType ?? this.dictType,
      dictCode: dictCode ?? this.dictCode,
      dictLabel: dictLabel ?? this.dictLabel,
      dictValue: dictValue ?? this.dictValue,
      dictDescription: dictDescription ?? this.dictDescription,
      cssClass: cssClass ?? this.cssClass,
      listClass: listClass ?? this.listClass,
      isDefault: isDefault ?? this.isDefault,
      isFixed: isFixed ?? this.isFixed,
      status: status ?? this.status,
      sortOrder: sortOrder ?? this.sortOrder,
      createdTime: createdTime ?? this.createdTime,
      modifiedTime: modifiedTime ?? this.modifiedTime,
      createdBy: createdBy ?? this.createdBy,
      updatedBy: updatedBy ?? this.updatedBy,
      deletedAt: deletedAt ?? this.deletedAt,
      remarks: remarks ?? this.remarks,
    );
  }

  factory SysDictModel.fromJson(Map<String, dynamic> json) {
    return SysDictModel(
      dictId: json['dictId'],
      parentId: json['parentId'],
      dictType: json['dictType'],
      dictCode: json['dictCode'],
      dictLabel: json['dictLabel'],
      dictValue: json['dictValue'],
      dictDescription: json['dictDescription'],
      cssClass: json['cssClass'],
      listClass: json['listClass'],
      isDefault: _parseBool(json['isDefault']),
      isFixed: _parseBool(json['isFixed']),
      status: json['status'],
      sortOrder: json['sortOrder'],
      createdTime: _parseDateTime(json['createdAt'] ?? json['createdTime']),
      modifiedTime: _parseDateTime(json['updatedAt'] ?? json['modifiedTime']),
      createdBy: json['createdBy'],
      updatedBy: json['updatedBy'],
      deletedAt: _parseDateTime(json['deletedAt']),
      remarks: json['remarks'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'dictId': dictId,
      'parentId': parentId,
      'dictType': dictType,
      'dictCode': dictCode,
      'dictLabel': dictLabel,
      'dictValue': dictValue,
      'dictDescription': dictDescription,
      'cssClass': cssClass,
      'listClass': listClass,
      'isDefault': isDefault,
      'isFixed': isFixed,
      'status': status,
      'sortOrder': sortOrder,
      'createdAt': createdTime?.toIso8601String(),
      'createdTime': createdTime?.toIso8601String(),
      'updatedAt': modifiedTime?.toIso8601String(),
      'modifiedTime': modifiedTime?.toIso8601String(),
      'createdBy': createdBy,
      'updatedBy': updatedBy,
      'deletedAt': deletedAt?.toIso8601String(),
      'remarks': remarks,
    };
  }

  static List<SysDictModel> listFromJson(List<dynamic> jsonList) {
    return jsonList
        .map((value) => SysDictModel.fromJson(value as Map<String, dynamic>))
        .toList();
  }

  static Map<String, SysDictModel> mapFromJson(
      Map<String, dynamic> json) {
    final map = <String, SysDictModel>{};
    if (json.isNotEmpty) {
      json.forEach((key, value) {
        map[key] =
            SysDictModel.fromJson(value as Map<String, dynamic>);
      });
    }
    return map;
  }

  static Map<String, List<SysDictModel>> mapListFromJson(
      Map<String, dynamic> json) {
    final map = <String, List<SysDictModel>>{};
    if (json.isNotEmpty) {
      json.forEach((key, value) {
        map[key] =
            SysDictModel.listFromJson(value as List<dynamic>);
      });
    }
    return map;
  }

  static DateTime? _parseDateTime(dynamic value) {
    if (value == null) return null;
    if (value is DateTime) return value;
    if (value is String && value.isNotEmpty) {
      return DateTime.tryParse(value);
    }
    return null;
  }

  static bool? _parseBool(dynamic value) {
    if (value == null) return null;
    if (value is bool) return value;
    if (value is num) return value != 0;
    if (value is String) {
      final lower = value.toLowerCase();
      if (lower == '1' || lower == 'true') return true;
      if (lower == '0' || lower == 'false') return false;
    }
    return null;
  }
}
